pkg/status/combinedstatus-resolution.go

```diff
@@ -1008,3 +1038,46 @@ func sortedStringSlice(s []string) []string {
 	sort.Sort(sort.StringSlice(s))
 	return s
 }
+
+// objectIsQueried checks whether `obj` appears in `query` as a standalone
+// occurrence.
+// That means, it must not be surrounded by alphanumeric characters.
+// query is assumed to be non-nil.
+func objectIsQueried(query *string, obj string) bool {
+	idx := 0
+
+	for {
+		idx = strings.Index((*query)[idx:], obj) // slices share the same storage, therefore efficient
+		if idx == -1 {
+			return false
+		}
+
+		if isWholeWord(query, idx, len(obj)) {
+			return true
+		}
+	}
+}
+
+// isWholeWord checks whether the word at `idx` in `s` is a whole word.
+// s is assumed to be non-nil.
+func isWholeWord(s *string, idx, length int) bool {
+	// check preceding rune
+	if idx > 0 {
+		r := rune((*s)[idx])
```

Comment:
This does not fetch the _preceding_ rune.

Reply:
Fixed.